plugins/funind/invfun.ml

```diff
@@ -38,38 +26,36 @@ let thin ids gl = Proofview.V82.of_tactic (Tactics.clear ids) gl
 
    if the type of hypothesis has not this form or if we cannot find the completeness lemma then we do nothing
 *)
-let revert_graph kn post_tac hid g =
-    let sigma = project g in
-    let typ = pf_unsafe_type_of g (mkVar hid) in
-    match EConstr.kind sigma typ with
-      | App(i,args) when isInd sigma i ->
-          let ((kn',num) as ind'),u = destInd sigma i in
-          if MutInd.equal kn kn'
-          then (* We have generated a graph hypothesis so that we must change it if we can *)
-            let info =
-              try find_Function_of_graph ind'
-              with Not_found -> (* The graphs are mutually recursive but we cannot find one of them !*)
-                anomaly (Pp.str "Cannot retrieve infos about a mutual block.")
-            in
-            (* if we can find a completeness lemma for this function
-               then we can come back to the functional form. If not, we do nothing
-            *)
-            match info.completeness_lemma with
-              | None -> tclIDTAC g
-              | Some f_complete ->
-                  let f_args,res = Array.chop (Array.length args - 1) args in
-                  tclTHENLIST
-                    [
-                      Proofview.V82.of_tactic (generalize [applist(mkConst f_complete,(Array.to_list f_args)@[res.(0);mkVar hid])]);
-                      thin [hid];
-                      Proofview.V82.of_tactic (Simple.intro hid);
-                      post_tac hid
-                    ]
-                    g
-
-          else tclIDTAC g
-      | _ -> tclIDTAC g
-
+let revert_graph kn post_tac hid gl =
```

Comment:
I think it is bad practice to separate a function entering a goal from the `enter` call. The preferred style in the codebase is instead
```
let foo bar quz = Proofview.Goal.enter begin fun gl ->
   ...
end
```

Reply:
Indeed, I agree, however that style is a blow in terms of indentation, all functions start at 4 spaces by default which hurts readability.